react/react_testing/introduction_to_react_testing.md

```diff
@@ -0,0 +1,208 @@
+### Introduction
+
+We've learned how to test our vanilla JavaScript applications in a previous section. Testing is indeed a powerful tool that allows us to write maintainable and flexible code. If you've followed along with our lessons so far, we've been using the [Jest](https://jestjs.io/) framework. Since, we're using [Vite](https://vitejs.dev/), we'll switch over to [Vitest](https://vitest.dev/) as our test runner since it integrates nicely with Vite. We'll add more capabilities to our tests using the [React Testing library (RTL)](https://testing-library.com/docs/react-testing-library/intro/).
```

Comment:
```suggestion
We've learned how to test our vanilla JavaScript applications in a previous section. Testing is a powerful tool that allows us to write maintainable and flexible code. If you've followed along with our lessons, so far we've been using the [Jest](https://jestjs.io/) framework. However, since we're using [Vite](https://vitejs.dev/) we'll switch over to [Vitest](https://vitest.dev/) as our test runner since it integrates nicely with Vite. We'll add more capabilities to our tests using the [React Testing library (RTL)](https://testing-library.com/docs/react-testing-library/intro/).
```

nit regarding `indeed`; feel free to disregard. The use of `indeed` just makes me feel like we were already in a conversation or something. But I'm not natively English either so might be quirky about things as well. Do whatever you feel is right with my suggestions :)

Reply:
Yeah I can see how `indeed` gives off that vibe. I think "already in a conversation" vibe is a good thing, no? It makes me feel like we are  little closer to the learner. However, I have no preference lol, if you feel strongly about this change, I'll change it.

Reply:
I don't feel strongly about it, but I might slightly prefer the bold statement `Testing is a powerful tool that allows us to write maintainable and flexible code`.

You did all the heavy lifting here so I really would like to leave it in your hands ☺️

Reply:
If you prefer it, then be it.